### PR TITLE
chore(iam-saml-provider): use commonAws in saml idp

### LINKED ISSUE #59: (Community::IAM::SamlProvider) Provider already exists
I'm using org-formation ver 0.9.13 trying to setup a SAML provider with `Community::IAM::SamlProvider` however I keep getting error `ERROR: Resource SamlProvider failed because Error: Provider with name testitorgmgnt-admin already exists..`

When i view the account with aws console there are no SAML providers setup in IAM.  Also I am able to manually go into AWS console -> IAM -> identity providers -> and add `testitorgmgnt-admin` SAML provider without error.

entire log execution: http://paste.openstack.org/show/801544/

org-formation repo is here: https://github.com/Sage-Bionetworks/testitorgmgnt-infra

### PR DESCRIPTION
**Pull request by OlafConijn**
chore: use commonAws in saml idp
fix: mark arn as 'createOnly' not 'readOnly'
closes  #59

### CODE REVIEW

**Comment by eduardomourar** [comment]
It should not be create only because those properties are expected to be filled that by the customer.

**Comment by OlafConijn** [comment]
> It should not be create only because those properties are expected to be filled that by the customer.

@eduardomourar ....then what should it be? this a bug in the rpdk?

**Comment by eduardomourar** [comment]
Not a bug. It just that we need to instantiate a new ResourceModel because the desired state is immutable

**Comment by OlafConijn** [comment]
right.... so that started when a dependency got updated where the desiredModel became immutable? 
maybe makes sense to update the errorMessage. i think this is the second time we got bit by it

**Comment by eduardomourar** [comment]
Yes, I believe that started on v0.3.0+. I will create a issue for it. FYI, I have made that clear in new the aws blog post.

File: iam/oidc-providers/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     },
     "dependencies": {
         "aws-resource-providers-common": "^0.2.0",
```
Comment (eduardomourar) [inline iam/oidc-providers/package.json:22]: this should be update to 0.3.0 and the lock file should be recreated as well

**Review by eduardomourar — CHANGES_REQUESTED** [review]

File: iam/saml-provider/package.json [hunk 2]
```diff
@@ -10,12 +10,17 @@
...
         "version": "npm run package && aws s3 cp ${npm_package_name}-${npm_package_version}.zip s3://community-resource-provider-catalog/${npm_package_name}-${npm_package_version}.zip && aws s3 cp resource-role.yaml s3://community-resource-provider-catalog/${npm_package_name}-resource-role-${npm_package_version}.yml",
-        "test": "npx jest"
+        "test": "npx jest --config ../../jest.config.js --rootDir .",
+        "test:integration": "npx npm-run-all -p -r samstart cfntest",
+        "cfntest": "cfn test -vv >> cfn.log",
+        "samstart": "sam local start-lambda -l sam.log",
+        "validate": "cfn validate"
     },
     "dependencies": {
+        "aws-resource-providers-common": "^0.2.0",
```
Comment (eduardomourar) [inline iam/saml-provider/package.json:23]: this should be 0.3.0 and lock file update

**Comment by eduardomourar** [comment]
the schema should also contain `replacementStrategy: delete_then_create` because that will be needed for most global resources

File: iam/oidc-providers/package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
...
-        "cfn-rpdk": "npm:@org-formation/cfn-rpdk@^0.4.0",
+        "cfn-rpdk": "npm:@org-formation/cfn-rpdk@^0.5.0",
         "class-transformer": "^0.3.1"
```
Comment (eduardomourar) [inline iam/oidc-providers/package.json:24]: i found an issue with the library `class-transformer` that you are probably facing. they introduced a breaking change. so we need to force the version we used:
```
"class-transformer": "0.3.1"
```

File: iam/saml-provider/package.json [hunk 4]
```diff
@@ -10,12 +10,17 @@
...
         "cfn-rpdk": "npm:@org-formation/cfn-rpdk@^0.5.0",
         "class-transformer": "^0.3.1"
```
Comment (eduardomourar) [inline iam/saml-provider/package.json:25]: same here:
```
"class-transformer": "0.3.1"
```

**Comment by OlafConijn** [comment]
> the schema should also contain `replacementStrategy: delete_then_create` because that will be needed for most global resources

not sure whether i follow. i assume with global you mean not-regional resources. what would the benefit be of delete_then_create? In my understanding delete_then_create makes it possible to rename the logical-id and global resources (1 per account) with a good default (no outage) would benefit from that, but i dont see that benefit here

**Comment by eduardomourar** [comment]
> > the schema should also contain `replacementStrategy: delete_then_create` because that will be needed for most global resources
> 
> not sure whether i follow. i assume with global you mean not-regional resources. what would the benefit be of delete_then_create? In my understanding delete_then_create makes it possible to rename the logical-id and global resources (1 per account) with a good default (no outage) would benefit from that, but i dont see that benefit here

yes, you are right. it makes sense to have that in the password policy. forget i said anything

**Review by eduardomourar — APPROVED** [review]
LGTM

**Pull request merged**